diff -up ipython-2.4.1/IPython/html/static/notebook/js/maintoolbar.js.fontawesome4 ipython-2.4.1/IPython/html/static/notebook/js/maintoolbar.js
--- ipython-2.4.1/IPython/html/static/notebook/js/maintoolbar.js.fontawesome4	2015-02-09 15:59:57.000000000 -0700
+++ ipython-2.4.1/IPython/html/static/notebook/js/maintoolbar.js	2015-05-08 16:59:45.309052483 -0600
@@ -27,7 +27,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'save_b',
                     label : 'Save and Checkpoint',
-                    icon : 'icon-save',
+                    icon : 'fa fa-save',
                     callback : function () {
                         IPython.notebook.save_checkpoint();
                         }
@@ -38,7 +38,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'insert_below_b',
                     label : 'Insert Cell Below',
-                    icon : 'icon-plus-sign',
+                    icon : 'fa fa-plus',
                     callback : function () {
                         IPython.notebook.insert_cell_below('code');
                         IPython.notebook.select_next();
@@ -51,7 +51,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'cut_b',
                     label : 'Cut Cell',
-                    icon : 'icon-cut',
+                    icon : 'fa fa-cut',
                     callback : function () {
                         IPython.notebook.cut_cell();
                         }
@@ -59,7 +59,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'copy_b',
                     label : 'Copy Cell',
-                    icon : 'icon-copy',
+                    icon : 'fa fa-copy',
                     callback : function () {
                         IPython.notebook.copy_cell();
                         }
@@ -67,7 +67,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'paste_b',
                     label : 'Paste Cell Below',
-                    icon : 'icon-paste',
+                    icon : 'fa fa-paste',
                     callback : function () {
                         IPython.notebook.paste_cell_below();
                         }
@@ -78,7 +78,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'move_up_b',
                     label : 'Move Cell Up',
-                    icon : 'icon-arrow-up',
+                    icon : 'fa fa-arrow-up',
                     callback : function () {
                         IPython.notebook.move_cell_up();
                         }
@@ -86,7 +86,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'move_down_b',
                     label : 'Move Cell Down',
-                    icon : 'icon-arrow-down',
+                    icon : 'fa fa-arrow-down',
                     callback : function () {
                         IPython.notebook.move_cell_down();
                         }
@@ -98,7 +98,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'run_b',
                     label : 'Run Cell',
-                    icon : 'icon-play',
+                    icon : 'fa fa-play',
                     callback : function () {
                         // emulate default shift-enter behavior
                         IPython.notebook.execute_cell_and_select_below();
@@ -107,7 +107,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'interrupt_b',
                     label : 'Interrupt',
-                    icon : 'icon-stop',
+                    icon : 'fa fa-stop',
                     callback : function () {
                         IPython.notebook.session.interrupt_kernel();
                         }
@@ -115,7 +115,7 @@ var IPython = (function (IPython) {
                 {
                     id : 'repeat_b',
                     label : 'Restart Kernel',
-                    icon : 'icon-repeat',
+                    icon : 'fa fa-repeat',
                     callback : function () {
                         IPython.notebook.restart_kernel();
                         }
diff -up ipython-2.4.1/IPython/html/static/notebook/less/notificationarea.less.fontawesome4 ipython-2.4.1/IPython/html/static/notebook/less/notificationarea.less
--- ipython-2.4.1/IPython/html/static/notebook/less/notificationarea.less.fontawesome4	2015-02-09 15:59:57.000000000 -0700
+++ ipython-2.4.1/IPython/html/static/notebook/less/notificationarea.less	2015-05-08 16:59:45.310052478 -0600
@@ -17,19 +17,19 @@
 }
 
 .edit_mode_icon:before {
-     .icon(@pencil);
+     content: @fa-var-pencil;
 }
 
 .command_mode_icon:before {
-     .icon(' ');
+     content: ' ';
 }
 
 .kernel_idle_icon:before {
-     .icon(@circle-blank);
+     content: @fa-var-circle-o;
 }
 
 .kernel_busy_icon:before {
-     .icon(@circle);
+     content: @fa-var-circle;
 }
 
 
diff -up ipython-2.4.1/IPython/html/static/style/style.less.fontawesome4 ipython-2.4.1/IPython/html/static/style/style.less
--- ipython-2.4.1/IPython/html/static/style/style.less.fontawesome4	2015-02-09 15:59:57.000000000 -0700
+++ ipython-2.4.1/IPython/html/static/style/style.less	2015-05-08 17:00:15.688893090 -0600
@@ -4,7 +4,7 @@
 
 // Font-Awesome
 @import "../components/font-awesome/less/font-awesome.less";
-@FontAwesomePath:    "../components/font-awesome/font";
+@fa-font-path:    "../components/font-awesome/font";
 
 // base
 @import "../base/less/style.less";
diff -up ipython-2.4.1/IPython/html/static/tree/less/tree.less.fontawesome4 ipython-2.4.1/IPython/html/static/tree/less/tree.less
--- ipython-2.4.1/IPython/html/static/tree/less/tree.less.fontawesome4	2015-02-09 15:59:57.000000000 -0700
+++ ipython-2.4.1/IPython/html/static/tree/less/tree.less	2015-05-08 16:59:45.310052478 -0600
@@ -128,9 +128,9 @@ input.engine_num_input {
 }
 
 .folder_icon:before {
-    .icon(@folder-close-alt)
+    content: @fa-var-folder-o
 }
 
 .notebook_icon:before {
-    .icon(@book)
+    content: @fa-var-book
 }
