From 0f57f9fe79fc8b5b070bb92c77dd1feb739c92bc Mon Sep 17 00:00:00 2001
From: Lumir Balhar <lbalhar@redhat.com>
Date: Wed, 20 May 2020 08:26:07 +0200
Subject: [PATCH] remove tests not compatible with python 3.9.0b1

---
 IPython/core/tests/test_oinspect.py | 32 -----------------------------
 1 file changed, 32 deletions(-)

diff --git a/IPython/core/tests/test_oinspect.py b/IPython/core/tests/test_oinspect.py
index eba51fd..f59c62f 100644
--- a/IPython/core/tests/test_oinspect.py
+++ b/IPython/core/tests/test_oinspect.py
@@ -381,35 +381,3 @@ def test_render_signature_short():
         short_fun.__name__,
     )
     nt.assert_equal(sig, 'short_fun(a=1)')
-
-
-def test_render_signature_long():
-    from typing import Optional
-
-    def long_function(
-        a_really_long_parameter: int,
-        and_another_long_one: bool = False,
-        let_us_make_sure_this_is_looong: Optional[str] = None,
-    ) -> bool: pass
-
-    sig = oinspect._render_signature(
-        signature(long_function),
-        long_function.__name__,
-    )
-    nt.assert_in(sig, [
-        # Python >=3.7
-        '''\
-long_function(
-    a_really_long_parameter: int,
-    and_another_long_one: bool = False,
-    let_us_make_sure_this_is_looong: Union[str, NoneType] = None,
-) -> bool\
-''',  # Python <=3.6
-        '''\
-long_function(
-    a_really_long_parameter:int,
-    and_another_long_one:bool=False,
-    let_us_make_sure_this_is_looong:Union[str, NoneType]=None,
-) -> bool\
-''',
-    ])
-- 
2.26.2

